src/controllers/swapAndBridge/swapAndBridge.ts

```diff
@@ -580,21 +585,23 @@ export class SwapAndBridgeController extends EventEmitter {
           let routeToSelect
           let routeToSelectSteps
 
-          const selectedRouteInQuoteRes = this.quote
-            ? quoteResult.routes.find((r: SocketAPIRoute) => {
-                if (this.quote!.selectedRoute?.usedBridgeNames) {
-                  return r?.usedBridgeNames?.[0] === this.quote!.selectedRoute.usedBridgeNames[0] // because we have only routes with unique bridges
-                }
-                if (this.quote!.selectedRoute?.usedDexName) {
-                  return r?.usedDexName === this.quote!.selectedRoute.usedDexName
-                }
-
-                return false
-              })
-            : null
-          if (selectedRouteInQuoteRes) {
-            routeToSelect = selectedRouteInQuoteRes
-            routeToSelectSteps = getQuoteRouteSteps(selectedRouteInQuoteRes.userTxs)
+          const alreadySelectedRoute = quoteResult.routes.find((nextRoute) => {
+            if (!this.quote) return false
+
+            // Because we have only routes with unique bridges (bridging case)
+            const selectedRouteUsedBridge = this.quote.selectedRoute.usedBridgeNames?.[0]
+            if (selectedRouteUsedBridge)
+              return nextRoute.usedBridgeNames?.[0] === selectedRouteUsedBridge
+
+            // Assuming to have routes with unique DEXes ONLY (swapping case)
+            const selectedRouteUsedDex = this.quote.selectedRoute.usedDexName
+            if (selectedRouteUsedDex) return nextRoute.usedDexName === selectedRouteUsedDex
```

Comment:
TODO: I'll double check this with Socket, because that's not always the case, see https://github.com/AmbireTech/ambire-app/issues/3173

Reply:
Yep, all good. Should be unique per quote. Example on Arbitrum:

<img width="782" alt="Screenshot 2024-11-07 at 11 53 13" src="https://github.com/user-attachments/assets/f54d84d4-b6ae-48ad-b817-523291487b80">